README.md

```diff
@@ -72,25 +72,23 @@ Depending on backend type, you might need Linux `sudo` to be used when running P
 ### Bluetooth Connection Options
 There is an optional parameter for `MoveHub` class constructor, accepting instance of `Connection` object. By default, it will try to use whatever `get_connection_auto()` returns. You have several options to manually control that:
 
-- use `pylgbst.get_connection_auto()` to attempt backend auto-choice, autodetect uses 
-- use `BlueGigaConnection()` - if you use BlueGiga Adapter (`pygatt` library prerequisite)
-- use `GattConnection()` - if you use Gatt Backend on Linux (`gatt` library prerequisite)
-- use `GattoolConnection()` - if you use GattTool Backend on Linux (`pygatt` library prerequisite)
-- use `GattLibConnection()` - if you use GattLib Backend on Linux (`gattlib` library prerequisite)
-- use `BluepyConnection()` - if you use Bluepy backend on Linux/Raspbian (`bluepy` library prerequisite)
-- use `BleakConnection()` - if you use Bleak backend (`bleak` library prerequisite)
+- use `pylgbst.get_connection_auto()` to attempt backend auto-detection 
+- use `pylgbst.get_connection_bluegiga()` - if you use BlueGiga Adapter (`pygatt` library prerequisite)
+- use `pylgbst.get_connection_gatt()` - if you use Gatt Backend on Linux (`gatt` library prerequisite)
+- use `pylgbst.get_connection_gattool()` - if you use GattTool Backend on Linux (`pygatt` library prerequisite)
+- use `pylgbst.get_connection_gattlib()` - if you use GattLib Backend on Linux (`gattlib` library prerequisite)
+- use `pylgbst.get_connection_bluepy()` - if you use Bluepy backend on Linux/Raspbian (`bluepy` library prerequisite)
+- use `pylgbst.get_connection_bleak()` - if you use Bleak backend (`bleak` library prerequisite)
 - pass instance of `DebugServerConnection` if you are using [Debug Server](#debug-server) (more details below).
 
 All the functions above have optional arguments to specify adapter name and Hub name (or mac address). Please take a look at functions source code for details.
 
 If you want to specify name for Bluetooth interface to use on local computer, you can pass that to class or function of getting a connection. Then pass connection object to `MoveHub` constructor. Like this:
 ```python
 from pylgbst.hub import MoveHub
-from pylgbst.comms.cgatt import GattConnection
```

Comment:
What exactly is wrong with this code? It supposed to work...

Reply:
Its not wrong, but it might be confusing to provide 2 ways to do the same thing. Looking at the GattConnection as an example, and changing it to BleakConnection took me on the wrong path that cost me quite some time, until I realized BleakConnection os not the one I need to be using, bu BleakDriver.  common get_connection_* helper interface sounds like a good idea, so I was actually surprised you implemented it but didn't push it here as the only way sanctioned. In fact, I would go further and simplify it even more, what do you think about this:
```
pylgbst.get_connection(backend: str = 'auto', controller: str ='hci0', hub_mac:str = None, hub_name:str = None)
```

Reply:
I understand your point about documentation offering different ways to connect and agree.

As for single `get_connection()` - I disagree. Relying on string constants for choosing the function that would work is quite implicit.

Reply:
doesnt't have to be a string, can be an enum or smth similar

Reply:
I understand, but that would be a different approach to organizing the library. This is more about readability and Zen of Python which states that `Explicit is better than implicit`.